test/jdk/java/nio/channels/FileChannel/Transfer2GPlus.java

```diff
@@ -115,28 +130,58 @@ private static void testToFileChannel(Path src, byte[] expected)
     // Exercises transferToArbitraryChannel() on all platforms.
     private static void testToWritableByteChannel(Path src, byte[] expected)
         throws IOException {
-        File file = File.createTempFile("dst", ".dat");
-        file.deleteOnExit();
-        try (FileChannel srcCh = FileChannel.open(src)) {
-            // The FileOutputStream is wrapped so that newChannel() does not
-            // return a FileChannelImpl and so make a faster path be taken.
-            try (DataOutputStream stream =
-                new DataOutputStream(new FileOutputStream(file))) {
-                try (WritableByteChannel wbc = Channels.newChannel(stream)) {
-                    long n;
-                    if ((n = srcCh.transferTo(0, LENGTH, wbc)) < LENGTH)
-                        throw new RuntimeException("Too few bytes transferred: " +
-                            n + " < " + LENGTH);
-
-                    System.out.println("Transferred " + n + " bytes");
-
-                    RandomAccessFile raf = new RandomAccessFile(file, "r");
-                    raf.seek(BASE);
-                    byte[] b = new byte[EXTRA];
-                    raf.read(b);
-                    if (!Arrays.equals(b, expected))
-                        throw new RuntimeException("Unexpected values");
-                }
+        // transfer src to channel that is not FileChannelImpl
+        try (FileChannel srcCh = FileChannel.open(src);
+             ByteArrayOutputStream baos = new ByteArrayOutputStream(EXTRA);
+             OutputStream os = new SkipBytesStream(baos, BASE);
+             WritableByteChannel wbc = Channels.newChannel(os)){
+
+            long n;
+            if ((n = srcCh.transferTo(0, LENGTH, wbc)) < LENGTH)
```

Comment:
The contract of `FileChannel::transferTo` does not guarantee that all bytes are transferred:

```An invocation of this method may or may not transfer all of the requested bytes [...]```.

Reply:
No, I think the docs should be left as they are.

Reply:
Ack that. If that's okay with you, I'd rather not change this code here; it was always like that, and I haven't found any related failure reports. The tests continue to pass.